src/linear_algebra/projective_space/basic.lean

```diff
@@ -92,19 +124,89 @@ lemma mk_eq_mk_iff (v w : V) (hv : v ≠ 0) (hw : w ≠ 0) :
   mk K v hv = mk K w hw ↔ ∃ (a : Kˣ), a • w = v :=
 quotient.eq'
 
+/-- A specialization of `mk_eq_mk_iff` that is sometimes more convenient to use. -/
+lemma mk_eq_mk_iff_mul_eq_mul {x y : K × K} (hx : x ≠ 0) (hy : y ≠ 0) :
+  mk K x hx = mk K y hy ↔ x.1 * y.2 = x.2 * y.1 :=
+begin
+  rw [mk_eq_mk_iff],
+  split,
+  { rintro ⟨a, rfl⟩,
+    simp [units.smul_def, mul_assoc, mul_comm y.1 _] },
+  { intro hxy,
+    rcases x with ⟨x1, x2⟩,
+    rcases y with ⟨y1, y2⟩,
+    rcases eq_or_ne y1 0 with (rfl | h),
+    { simp only [ne.def, prod.mk_eq_zero, eq_self_iff_true, true_and] at hy,
+      simp only [hy, mul_zero, mul_eq_zero, or_false] at hxy,
+      simp only [hxy, ne.def, prod.mk_eq_zero, eq_self_iff_true, true_and] at hx,
+      use units.mk0 (x2/y2) (div_ne_zero hx hy),
+      simp [units.smul_def, hy, hxy] },
+    { rcases eq_or_ne x1 0 with (rfl | h'),
+      { simp only [ne.def, prod.mk_eq_zero, eq_self_iff_true, true_and] at hx,
+        simp only [hx, h, zero_mul, zero_eq_mul, false_or] at hxy,
+        contradiction },
+      { use units.mk0 (x1/y1) (div_ne_zero h' h),
+        simp [units.smul_def, div_mul_cancel, h, mul_comm_div', ← mul_div_assoc, hxy] } } }
+end
+
 lemma exists_smul_eq_mk_rep
   (v : V) (hv : v ≠ 0) : ∃ (a : Kˣ), a • v = (mk K v hv).rep :=
 show (projectivization_setoid K V).rel _ _, from quotient.mk_out' ⟨v, hv⟩
 
 variable {K}
 
+/-- Define a function on the projective line from a function of the ratio of projective
+coordinates. -/
+def lift_of_div {α K : Type*} [field K] (f : K → α) (z : ℙ K (K × K)) : α :=
```

Comment:
I'm not sure about this definition. It is taking advantage of the fact that we define division-by-zero (using junk value zero).

As a result the function on the projective line that `lift_of_div` constructs always gives the same value to `0` as to `∞`:
```lean
example {α : Type*} (f : K → α) :
  lift_of_div f (mk K (1, 0) (by simp)) = lift_of_div f (mk K (0, 1) (by simp)) :=
by simp [lift_of_div]
```

Consider its behaviour if we feed in the function $x ↦ [x, 1] : K → Kℙ¹$, or in Lean `lift_of_div (λ (x : K), mk K (x, (1 : K)) (by simp))`. In this case, `lift_of_div` creates a function that is the identity everywhere except at `∞`.

Is this definitely a useful construction?